app/android/app_template/BUILD.gn

```diff
@@ -0,0 +1,67 @@
+import("//build/config/android/rules.gni")
+
+android_apk("xwalk_app_template_apk") {
+  apk_name = "XWalkAppTemplate"
+  android_manifest = "AndroidManifest.xml"
+  java_files = [ "src/org/xwalk/app/template/AppTemplateActivity.java" ]
+  deps = [
+    ":xwalk_app_template_apk_assets",
+    ":xwalk_app_template_apk_resources",
+    "//xwalk/app/android/runtime_client:xwalk_app_runtime_java",
+
+    #TODO(heke123): remove xwalk_core_java, it should be embedded in xwalk_app_runtime_java.
+    "//xwalk/runtime/android/core:xwalk_core_java",
+  ]
+}
+
+android_assets("xwalk_app_template_apk_assets") {
+  sources = []
+  renaming_sources = [
+    "//xwalk/test/android/data/index.html",
+    "//xwalk/test/android/data/sampapp-icon-helloworld.png",
+  ]
+  renaming_destinations = [
+    "www/index.html",
+    "www/sampapp-icon-helloworld.png",
+  ]
+  disable_compression = true
+}
+
+android_resources("xwalk_app_template_apk_resources") {
+  resource_dirs = [ "res" ]
+  custom_package = "org.xwalk.app.template"
+}
+
+action("prepare_xwalk_app_template") {
+  script = "//xwalk/build/android/gn/generate_app_packaging_tool.py"
```

Comment:
another unsure point here:  I add the "//xwalk/build/android/gn/generate_app_packaging_tool.py", which is slightly different with original "//xwalk/build/android/generate_app_packaging_tool.py".
I remember last time you mentioned that the new created scripts in gn/ folder is not acceptable, any suggestions?